platforms/BUILD

```diff
@@ -1,25 +1,36 @@
 package(default_visibility = ["//visibility:public"])
 
-config_setting(
+platform(
     name = "linux_x86_64",
     constraint_values = [
-        "@platforms//os:linux",
         "@platforms//cpu:x86_64",
+        "@platforms//os:linux",
     ],
+    exec_properties = {
+        "OSFamily": "Linux",
+        "container-image": "docker://gcr.io/flame-public/rbe-ubuntu20-04:latest",
+        "dockerNetwork": "off",
+    },
 )
 
-config_setting(
-    name = "macos_x86_64",
+platform(
+    name = "test_with_privileged",
     constraint_values = [
-        "@platforms//os:macos",
-        "@platforms//cpu:x86_64",
+        "//platforms/configs:with_privileged",
     ],
+    exec_properties = {
+        "test.workload-isolation-type": "firecracker",
+    },
+    parents = ["//platforms:linux_x86_64"],
 )
 
-config_setting(
-    name = "macos_arm64",
+platform(
+    name = "test_with_docker_daemon",
     constraint_values = [
-        "@platforms//os:macos",
-        "@platforms//cpu:arm64",
+        "//platforms/configs:with_docker_daemon",
     ],
+    exec_properties = {
+        "test.init-dockerd": "true",
+    },
+    parents = ["//platforms:test_with_privileged"],
 )
```

Comment:
I think maybe the `platform` name should describe what the concrete platform actually is, not the constraints it satisfies. e.g. you'd say "I'm running this test on `linux_x86_64`" or "I'm running on the `macos_x86_64` platform" but not "I'm running this test on `test_with_privileged`".

So I feel like these should be called `firecracker_x86_64_test` and `firecracker_x86_64_dockerd_test`